src/main/java/teammates/common/util/Config.java

```diff
@@ -132,7 +131,22 @@ static String getBaseAppUrl() {
      * Returns true if the server is configured to be the dev server.
      */
     public static boolean isDevServer() {
-        return SystemProperty.environment.value() != SystemProperty.Environment.Value.Production;
+        // In production server, GAE sets some non-overrideable environment variables.
+        // We will make use of some of them to determine whether the server is dev server or not.
+        // This means that any developer can replicate this condition in dev server,
+        // but it is their own choice and risk should they choose to do so.
+
+        String appName = System.getenv("GAE_APPLICATION");
+        String version = System.getenv("GAE_VERSION");
+        String env = System.getenv("GAE_ENV");
+
+        if (appName == null || version == null || env == null) {
+            return true;
+        }
+
+        return !appName.endsWith(APP_ID)
+                || !version.replace("-", ".").equals(APP_VERSION)
+                || !"standard".equals(env);
```

Comment:
`!env.equals("standard")` seems to be a more consistent style?

Reply:
Not really. Literals are usually placed on the left to prevent NPE. So in this case I modify the line just above this.